### DISCUSSION TITLE
Efergy sensor , Weewx & weewx-GTS

### DISCUSSION THREAD

**Discussion by MksRasp** [roe-dl/weewx-GTS#17] [Q&A] [answered]
we have asked for help from weewx-user discussions group about efergy sensor and how to benefit from the data this sensor is spitting and put it in use like dialy , weekly, monthly consumption and cost. they suggested to install weewx-GTS which we did but we got glued because we do not know how to configure it.

our hardware consists of:

1-raspberry pi 4 running bullseye with no issue.
2-SDR dongle capturing Acurite weather station data and also the data of efergy sensor.( rtl_433 and rtl-sdr can send **"Amps"** or **"kw"** to weewx but we selected **"kw"** and it is logged in weewx database under the name of **"electricity"** every 5 minutes)
3-weewx v4.8.0 (Python version 3.9.2) and seasons skin.
4-weewx-GTS

the efergy kit came with LCD display and its reading in **"kw"** and it always in agreement with captured data by SDR dongle which is repeated every 20 seconds.

thank you

**Comment by roe-dl** [comment]
Which skin do you use?

Did you learn how to customize that skin?

**Comment by MksRasp** [comment]
seasons skin and we can work on customization but not with python codes

**Comment by MksRasp** [comment]
we want to test weewx-GTS on VirtualBox running Linux Mint 19.3 Cinnamon.
we could not install weewx python3 on this test machine but we were able to install weewx python2 version and it is running just fine.
we then installed weewx-GTS and the installation finished without problem. 
stopped weewx and start it again but unfortunately weewx crashed with the following error

Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****    File "/usr/share/weewx/weewx/engine.py", line 93, in __init__
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****      self.loadServices(config_dict)
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****    File "/usr/share/weewx/weewx/engine.py", line 161, in loadServices
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****      obj = weeutil.weeutil.get_object(svc)(self, config_dict)
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****    File "/usr/share/weewx/weeutil/weeutil.py", line 1150, in get_object
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****      mod = __import__(module)
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****    File "/usr/share/weewx/user/GTS.py", line 17
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****  SyntaxError: Non-ASCII character '\xc2' in file /usr/share/weewx/user/GTS.py on line 18, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
Sep  9 20:19:07 lnx-VirtualBox weewx[2015] CRITICAL __main__:     ****  Exiting.


how we can fix this error?

**Comment by roe-dl** [comment]
Are you sure you use Python 3? That error message sounds like Python 2

**Reply by MksRasp** [reply]
> Are you sure you use Python 3? That error message sounds like Python 2

I have 2 system:
system 1 (virtual machine on windows 10 pc), this system have **python2** weewx and weewx-GTS installed fine but when weewx run it crashed with this error (SyntaxError: Non-ASCII character '\xc2' in file /usr/share/weewx/user/GTS.py on line 18, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details)

system 2 (raspberry pi) this system have python3 weewx and weewx-GTS installed fine and running fine but we are facing some issue with units that is showing watts but we want kilowatt instead,

we need your help with units issue on system 2 if possible

thank you

**Comment by MksRasp** [comment, marked as answer]
got it fixed

thank you
